src/utils/token.rs

```diff
@@ -32,7 +30,18 @@ use crate::model::id::UserId;
 /// Returns a [`InvalidToken`] when one of the above checks fail.
 /// The type of failure is not specified.
 pub fn validate(token: impl AsRef<str>) -> Result<(), InvalidToken> {
-    parse(token).map(|_| ()).ok_or(InvalidToken)
+    // Tokens can be preceded by "Bot " (that's how the Discord API expects them)
+    let mut parts = token.as_ref().trim_start_matches("Bot ").split('.');
+
+    let is_valid = parts.next().map_or(false, |p| !p.is_empty())
+        && parts.next().map_or(false, |p| !p.is_empty())
+        && parts.next().map_or(false, |p| !p.is_empty())
+        && parts.next().is_none();
+
+    match is_valid {
```

Comment:
Could you fix the clippy warning here?

Reply:
Ah sadge we lint against that. I prefer the match because it's shorter and aligns the branches which is more pleasant to read. But oh well

Also I wonder why the clippy CI doesn't catch this. In general I feel like our CI is pretty leaky with regards to warnings